torch/distributed/_composable/fully_shard.py

```diff
@@ -45,6 +45,7 @@ def fully_shard(
     device_id: Optional[Union[int, torch.device]] = None,
     param_init_fn: Optional[Callable[[nn.Module], None]] = None,
     sync_module_states: bool = False,
+    limit_all_gathers: bool = True,
```

Comment:
Would we want to expose other flags, such as backward / forward prefetch, etc as well or are we confident in these defaults?